pkg/controller/installation/validation.go

```diff
@@ -145,6 +148,37 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 		}
 	}
 
+	switch instance.Spec.CNI.Type {
+	case operatorv1.PluginCalico:
+	case operatorv1.PluginGKE:
+		switch instance.Spec.KubernetesProvider {
+		case operatorv1.ProviderEKS:
+			fallthrough
+		case operatorv1.ProviderAKS:
+			return fmt.Errorf("kubernetesProvider of %s is invalid with CNI.type %s",
+				instance.Spec.KubernetesProvider, instance.Spec.CNI.Type)
```

Comment:
Neat. I've never seen a `fallthrough` before.

If I understand it correctly, doesn't this do the same as the following (in an arguably less-clear way?)
```suggestion
		case operatorv1.ProviderEKS, operatorv1.ProviderAKS:
			return fmt.Errorf("kubernetesProvider of %s is invalid with CNI.type %s",
				instance.Spec.KubernetesProvider, instance.Spec.CNI.Type)
```

Reply:
Right, your way is clearer. Sorry I think my C++ background is showing.